standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStoreClient.java

```diff
@@ -2245,8 +2245,8 @@ public List<Partition> listPartitions(String db_name, String tbl_name,
   public List<Partition> listPartitions(String catName, String db_name, String tbl_name,
                                         List<String> part_vals, int max_parts) throws TException {
     // TODO should we add capabilities here as well as it returns Partition objects
-    if (db_name == null || tbl_name == null || part_vals == null) {
-      throw new MetaException("Database name/Table name/partition values should not be null");
+    if (db_name == null || tbl_name == null) {
```

Comment:
nit: if the client invokes this method specifically, I would like to add a check `part_vals == null`

Reply:
Since metastore server supports null and empty partial values, I think client can also support it.

Reply:
I think the `part_vals` shouldn't be null according to the comment indicates: `Get a list of partitions based on a (possibly partial) list of partition values`.
Another point is what would happen if the new client connects to the old server which hasn't upgraded?

Reply:
Although old server supports null part_vals, let's revert to the original client behavior.